pkg/forwarder/blocked_endpoints.go

```diff
@@ -26,35 +28,54 @@ type blockedEndpoints struct {
 }
 
 func newBlockedEndpoints() *blockedEndpoints {
+	// All forwarder settings are used directly or indirectly (GetBackoffDuration) by
+	// this circuit breaker singleton. Therefore, it makes sense to load them here.
+	if !configLoaded {
```

Comment:
I think we should move the code in `pkg/forwarder/backoff.go` in the `blockerEndpoints` struct. That would avoid having to check if we have already initialized the constant (and get rid of `configLoaded`). Also it remove the init part where we load the default values from the config to overwrite them later and it simplify the code in my opinion.

Reply:
We use the global vars mostly in `GetBackoffDuration`:

https://github.com/DataDog/datadog-agent/pull/1458/files#diff-f5d8ee754e1fc4223fc20e7e6bdd28fdR94

Storing the settings in `blockedEndpoints` instead would require passing an instance of that to the backoff function which would result in too tight of a coupling imo (and would necessitate changing much of this PR).

Also, for testing we also use `loadConfig` heavily to reset default values:

https://github.com/DataDog/datadog-agent/pull/1458/commits/9e01e83b4e78553e11a7a590baf463c21a6d60a8#diff-8e4dcd771e37bb4ab78485e4a92a717a

Reply:
My point was to make functions from `pkg/forwarder/backoff.go` methods of `blockedEndpoints`. It would simplify the code and testing and change almost nothing in the PR since no functions or constants from `backoff.go` are used anywhere else than inside `blockedEndpoints`.

For testing we would remove `loadConfig` entirely and get the value from the configuration from `newBlockedEndpoints`. This is a pattern we use everywhere else in the agent and is better, in my opinion, than having a constant and global to control initialization.